main/src/vocab/nllb_vocab.rs

```diff
@@ -0,0 +1,245 @@
+    /// values), special values typically include things like BOS/EOS markers, class markers, mask
+    /// markers and padding markers
+    pub special_values: HashMap<String, i64>,
+
+    /// A mapping of special value tokens as IDs to strings (i.e. the decoder base for special values)
+    pub special_indices: HashMap<i64, String>,
+
+    /// Language code stored as bytes for extraction of the prefix in input sequences
+    pub language_codes_bytes: HashSet<Vec<u8>>,
+
+    pub special_token_storage: SpecialTokenMap,
+}
+
+impl NLLBVocab {
+    /// The beginning of sequence token that was used during pretraining.
+    /// Can be used a sequence classifier token.
+    pub fn get_bos_value(&self) -> &str {
+        self.special_token_storage
+            .bos_token
+            .as_deref()
+            .unwrap_or("<s>")
+    }
+
+    /// End of sequence token.
+    pub fn get_eos_value(&self) -> &str {
+        self.special_token_storage
+            .eos_token
+            .as_deref()
+            .unwrap_or("</s>")
+    }
+
+    /// Returns the SEP token for M2M100 (`</s>`)
+    pub fn get_sep_value(&self) -> &str {
+        self.special_token_storage
+            .eos_token
+            .as_deref()
+            .unwrap_or("</s>")
+    }
+
+    /// Returns the PAD token for M2M100 (`<pad>`)
+    pub fn get_pad_value(&self) -> &str {
+        self.special_token_storage
+            .pad_token
+            .as_deref()
+            .unwrap_or("<pad>")
+    }
+}
+
+impl Vocab for NLLBVocab {
+    fn get_unknown_value(&self) -> &str {
+        self.special_token_storage.unk_token.as_str()
+    }
+
+    fn values(&self) -> &HashMap<String, i64> {
+        &self.values
+    }
+
+    fn indices(&self) -> &HashMap<i64, String> {
+        &self.indices
+    }
+
+    fn special_values(&self) -> &HashMap<String, i64> {
+        &self.special_values
+    }
+
+    fn special_indices(&self) -> &HashMap<i64, String> {
+        &self.special_indices
+    }
+
+    fn from_file<V: AsRef<Path>>(vocab: V) -> Result<Self, TokenizerError> {
+        let tokenizer = Tokenizer::deserialize(vocab)?;
+        let special_config = SpecialTokenMap {
+            unk_token: "<unk>".into(),
+            ..Default::default()
+        };
+
+        Self::from_values_and_special_token_map(tokenizer.model.vocab, special_config)
+    }
+
+    fn token_to_id(&self, token: &str) -> i64 {
+        self._token_to_id(
+            token,
+            self.values(),
+            self.special_values(),
+            self.get_unknown_value(),
+        )
+    }
+
+    fn id_to_token(&self, id: &i64) -> String {
+        self._id_to_token(
+            id,
+            self.indices(),
+            self.special_indices(),
+            self.get_unknown_value(),
+        )
+    }
+
+    fn from_file_with_special_token_mapping<P: AsRef<Path>, S: AsRef<Path>>(
+        path: P,
+        special_token_mapping_path: S,
+    ) -> Result<Self, TokenizerError>
+    where
+        Self: Sized,
```

Comment:
is the `Sized` specification required here?

Reply:
You're right, it's redundant. Fixed in https://github.com/npatsakula/rust-tokenizers/pull/5.